### PR TITLE
Add properties for DatetimeIndex

### PR DESCRIPTION
**Pull request by xinrong-databricks**
The PR is proposed to add properties for DatetimeIndex. The added properties follow datetimelike properties of the Series values.

### CODE REVIEW

File: databricks/koalas/missing/indexes.py [hunk 1]
```diff
@@ -100,35 +100,13 @@ class MissingPandasLikeIndex(object):
...
 
     # Properties
-    year = _unsupported_property("year", cls="DatetimeIndex")
-    month = _unsupported_property("month", cls="DatetimeIndex")
-    day = _unsupported_property("day", cls="DatetimeIndex")
-    hour = _unsupported_property("hour", cls="DatetimeIndex")
-    minute = _unsupported_property("minute", cls="DatetimeIndex")
-    second = _unsupported_property("second", cls="DatetimeIndex")
-    microsecond = _unsupported_property("microsecond", cls="DatetimeIndex")
     nanosecond = _unsupported_property("nanosecond", cls="DatetimeIndex")
```
Comment (xinrong-databricks) [inline databricks/koalas/missing/indexes.py:103]: Properties that are still missing would be implemented in follow-up PRs.

File: databricks/koalas/tests/test_indexes.py [hunk 2]
```diff
@@ -370,6 +370,44 @@ def test_index_unique(self):
...
+            self.assert_eq(kid.quarter, pid.quarter)
+            self.assert_eq(kid.daysinmonth, pid.daysinmonth)
+            self.assert_eq(kid.days_in_month, pid.days_in_month)
+            self.assert_eq(kid.is_month_start, ks.Index(pid.is_month_start))
+            self.assert_eq(kid.is_month_end, ks.Index(pid.is_month_end))
+            self.assert_eq(kid.is_quarter_start, ks.Index(pid.is_quarter_start))
+            self.assert_eq(kid.is_quarter_end, ks.Index(pid.is_quarter_end))
+            self.assert_eq(kid.is_year_start, ks.Index(pid.is_year_start))
+            self.assert_eq(kid.is_year_end, ks.Index(pid.is_year_end))
+            self.assert_eq(kid.is_leap_year, ks.Index(pid.is_leap_year))
```
Comment (ueshin) [inline databricks/koalas/tests/test_indexes.py:405]: `pd.Index` instead of `ks.Index` for expected values?

**Review by ueshin — COMMENTED** [review]
Could you also update `docs/source/reference/indexing.rst`?

File: databricks/koalas/tests/test_indexes.py (hunk 2 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_indexes.py:405]: Makes sense!

**Comment by xinrong-databricks** [comment]
> Could you also update `docs/source/reference/indexing.rst`?

Sounds good! Updated.

File: docs/source/reference/indexing.rst [hunk 3]
```diff
@@ -293,3 +293,33 @@ DatatimeIndex
    :toctree: api/
 
    DatetimeIndex
+
+Properties
```
Comment (ueshin) [inline docs/source/reference/indexing.rst:297]: `Time/date components`? https://pandas.pydata.org/pandas-docs/stable/reference/indexing.html#time-date-components

File: databricks/koalas/tests/test_indexes.py [hunk 4]
```diff
@@ -370,6 +370,44 @@ def test_index_unique(self):
         with self.assertRaisesRegex(KeyError, "Requested level (hi)*"):
             kidx.unique(level="hi")
 
+    def test_datetime_index_properties(self):
+        pids = [
```
Comment (ueshin) [inline databricks/koalas/tests/test_indexes.py:374]: nit: `pidxs`?

File: databricks/koalas/tests/test_indexes.py [hunk 5]
```diff
@@ -370,6 +370,44 @@ def test_index_unique(self):
...
+            pd.DatetimeIndex([0]),
+            pd.DatetimeIndex(["2004-01-01", "2002-12-31", "2000-04-01"]),
+        ] + [
+            pd.date_range("2000-01-01", periods=3, freq=unit)
+            for unit in ["ns", "us", "ms", "s", "m", "h", "D"]
+        ]
+
+        for pid in pids:
```
Comment (ueshin) [inline databricks/koalas/tests/test_indexes.py:382]: nit: `pidx`?

File: databricks/koalas/tests/test_indexes.py [hunk 6]
```diff
@@ -370,6 +370,44 @@ def test_index_unique(self):
...
+            kid = ks.from_pandas(pid)
```
Comment (ueshin) [inline databricks/koalas/tests/test_indexes.py:383]: nit: `kidx`?

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM.

File: docs/source/reference/indexing.rst (hunk 3 above)
Comment (xinrong-databricks) [reply docs/source/reference/indexing.rst]: Good catch, updated!

File: databricks/koalas/tests/test_indexes.py (hunk 4 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_indexes.py]: Modified.

File: databricks/koalas/tests/test_indexes.py (hunk 5 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_indexes.py]: Modified.

File: databricks/koalas/tests/test_indexes.py (hunk 6 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_indexes.py]: Modified.

**Comment by xinrong-databricks** [comment]
Merged! Thank you!

**Pull request merged**